lib/model/CEventRateModel.cc

```diff
@@ -322,12 +324,23 @@ void CEventRateModel::sample(core_t::TTime startTime,
                                   priorWeights[0]);
                 maths_t::setWinsorisationWeight(winsorisationWeight, priorWeights[0]);
 
+                const auto createAndAddAnnotation = [&](core_t::TTime t,
+                                                        const std::string& annotation) {
+                    m_CurrentBucketStats.s_Annotations.emplace_back(
+                        t, annotation, gatherer.searchKey().detectorIndex(),
+                        gatherer.searchKey().partitionFieldName(),
+                        gatherer.partitionFieldValue(),
+                        gatherer.searchKey().overFieldName(), "",
```

Comment:
You could use `EMPTY_STRING` instead of `""`.  It's a static constant in the base class and avoids a call to `strlen` on the `const char*`.